example/invariance.prl

```diff
@@ -0,0 +1,184 @@
+      claim p/1 : (@ #p 1) = %r in %ty by {auto};
+      (#t p/0 p/1)
+    ]
+  }
+}.
+
+theorem FunToPairIsEquiv :
+  (->
+   [ty : (U 0 kan)]
+   (IsEquiv (-> bool ty) (* ty ty) ($ FunToPair ty)))
+by {
+  lam ty pair =>
+  { { lam b => if b then `(!proj1 pair) else `(!proj2 pair)
+    , abs _ => `pair }
+  , unfold Fiber;
+    lam {fun,p} =>
+     (GetEndpoints p [p/0 p/1] #tac{
+      (abs x =>
+        {lam b => if b then `(!proj1 (@ p x)) else `(!proj2 (@ p x)),
+         abs y =>
+           `(@ ($ (WeakConnection #lvl{0}) (* ty ty) ($ FunToPair ty fun) pair p) x y)
+        });
+      [ unfold FunToPair in p/0; reduce in p/0 at right;
+        inversion; with q3 q2 q1 q0 =>
+          reduce at right in q2;
+          reduce at right in q3;
+          auto; with b =>
+            elim b; reduce at right; symmetry; assumption
+      , unfold FunToPair in p/1; reduce in p/1 at right;
+        inversion; with q3 q2 q1 q0 => elim pair;
+        reduce at right in q0; reduce at right in q1;
+        auto; assumption
+      ]
+     })
+  }
+}.
+
+// }}}
+
+// By univalence, there is a path between these two types.
+theorem FunEqPair :
+  (->
+   [ty : (U 0 kan)]
+   (path [_] (U 0 kan) (-> bool ty) (* ty ty)))
+by {
+  lam ty => abs x =>
+  `(V x (-> bool ty) (* ty ty)
+    (tuple [proj1 ($ FunToPair ty)] [proj2 ($ FunToPairIsEquiv ty)]))
+}.
+
+// We can coerce functions to pairs, and this coercion will compute.
+theorem CoerceFunToPair :
+  (->
+   [ty : (U 0 kan)]
+   (-> bool ty)
+   (* ty ty))
+by {
+  lam ty fun =>
+  `(coe 0~>1 [x] (@ ($ FunEqPair ty) x) fun)
+}.
+
+theorem ComputeCoercion :
+  (=
+   (* bool bool)
+   ($ CoerceFunToPair bool (lam [b] b))
+   (tuple [proj1 tt] [proj2 ff]))
+by {
+  auto
+}.
+
+// We can define a function on pairs, coerce it to a function on functions, and
+// this coercion will compute.
+theorem SwapPair :
+  (->
+    [ty : (U 0 kan)]
+    (* ty ty)
+    (* ty ty))
+by {
+  lam ty {p1,p2} => {`p2,`p1}
+}.
+
+define SwapCoe(#ty,#r:dim) =
+  (coe 1~>#r [x] (-> (@ ($ FunEqPair #ty) x) (@ ($ FunEqPair #ty) x)) ($ SwapPair #ty)).
+
+theorem SwapFun :
+  (->
+    [ty : (U 0 kan)]
+    (-> bool ty)
+    (-> bool ty))
+by {
+  lam ty => `(SwapCoe ty 0)
+}.
+
+theorem ComputeSwap :
+  (=
+    bool
+    ($ SwapFun bool (lam [b] b) tt)
+    ff)
+by {
+  auto
+}.
+
+// We can prove that SwapPair o SwapPair = id, and coerce this to the same
+// equation on SwapFun.
+theorem SwapPairEqn :
+  (->
+    [ty : (U 0 kan)]
+    [pair : (* ty ty)]
+    (path [_] (* ty ty) ($ SwapPair ty ($ SwapPair ty pair)) pair))
+by {
+  lam ty pair => abs x => `pair
+}.
+
+theorem SwapFunEqn :
+  (->
+    [ty : (U 0 kan)]
+    [fun : (-> bool ty)]
+    (path [_] (-> bool ty) ($ SwapFun ty ($ SwapFun ty fun)) fun))
+by {
+  lam ty =>
+  `(coe 1~>0
+      [x] (-> [elt : (@ ($ FunEqPair ty) x)]
+              (path [_] (@ ($ FunEqPair ty) x)
+                    ($ (SwapCoe ty x) ($ (SwapCoe ty x) elt))
```

Comment:
I don't like this indentation. `($...` should have its indentation not determined by the right end of `path`, but by the left end of that lexeme.